java/core/src/java/org/apache/orc/impl/HadoopShims.java

```diff
@@ -124,6 +126,43 @@
    */
   public TextReaderShim getTextReaderShim(InputStream input) throws IOException;
 
+  
+  /** 
+   * Block filler shim - make sure the DFS blocks ends at this offset.
+   */
+  public interface BlockFillerShim {
```

Comment:
I think it would be better to create a HadoopShims_2_5 that has the simple fill method and make HadoopShimsCurrent have the full featured one.